snitun/client/client_peer.py

```diff
@@ -41,7 +41,9 @@ def wait(self) -> asyncio.Task:
         """Block until connection to peer is closed."""
         if not self._multiplexer:
             raise RuntimeError("No SniTun connection available")
-        return self._multiplexer.wait()
+        # Wait until the handler task is done
+        # as we know the connection is closed
+        return make_task_waiter_future(self._handler_task)
```

Comment:
It would have been nicer if we could simply await `asyncio.wait` here but we need to return something that waits on the task and prevents the state from leaking upward.